### PR TITLE
Improvement/linters

### PR DESCRIPTION
**Pull request by ethan605**
# Problem

The current project's linter configurations prevent some rules of the new [Readability & Maintainability guideline](https://wiki.onfido.net/pages/viewpage.action?spaceKey=SDK&title=Development+guideline+for+Web+SDK#DevelopmentguidelineforWebSDK-c.Readability&Maintainability) to be effective.

# Solution

To integrate & strictly follow recommended rules:

- `eslint:recommended`
- `plugin:react/recommended`
- `preact`
- `plugin:flowtype/recommended`
- `plugin:prettier/recommended`
- `prettier/standard`

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [n/a] Has the CHANGELOG been updated?
- [n/a] Has the README been updated?
- [n/a] Has the CONTRIBUTING doc been updated?
- [n/a] Has the RELEASE_GUIDELINES been updated?
- [n/a] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [n/a] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [n/a] Have any new automated tests been implemented or the existing ones changed?
- [n/a] Have any new manual tests been written down or the existing ones changed?
- [n/a] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1156-pr-onfido-sdk-ui-onfido.surge.sh

**Review by kopijunkie — APPROVED** [review]
Looks good. Any particular reason for switching from using a `.eslintrc` config to `.eslintrc.js`?

**Comment by ethan605** [comment]
> Looks good. Any particular reason for switching from using a `.eslintrc` config to `.eslintrc.js`?

There're 2 main reasons:
- Natural readability for all IDEs
- Easier to manipulate contents, e.g. documenting, quickly toggling rules, etc.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    },
+  },
+  plugins: ['react', 'flowtype', 'mocha', 'prettier'],
+  globals: {
+    expect: false,
+  },
+  rules: {
+    /**
+     * Temporary disabled rules for CI passing
+     * @TODO merge improvement/linters-fixes PR
```
Comment (ethan605) [inline .eslintrc.js:31]: @kopijunkie inline documents like this

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1156-pr-onfido-sdk-ui-onfido.surge.sh

File: .eslintrc.js (hunk 1 above)
Comment (kopijunkie) [reply .eslintrc.js:31]: Ah, yes. Good point about inline documentation, since JSON doesn't allow for that.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Review by stefaniacardenas — COMMENTED** [review]
There is also an`.eslintrc` in the `/test` folder. Does it still make sense to have two different linter configurations? Can we merge them or apply similar rules?

**Comment by ethan605** [comment]
> There is also an`.eslintrc` in the `/test` folder. Does it still make sense to have two different linter configurations? Can we merge them or apply similar rules?

Yes we can safely remove that file because it's redundant.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1156-pr-onfido-sdk-ui-onfido.surge.sh

**Review by stefaniacardenas — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1156-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1156-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**